### DISCUSSION TITLE
(No backend) In browser only template/example

### DISCUSSION THREAD

**Discussion by mehdi-cit** [eclipse-glsp/glsp#1075] [General]
Hello everyone.
While listening to Philip Langer video [here](https://youtu.be/f5gVKh04Oho?t=4638) (1:17:18) and surely he spoke about the possibility of running everything in the browser (no backend on a different machine/process).
Problem is that I am not able to locate any consistent doc about this scenario. A template or a YEOMAN generator would be much useful. 
Also, I think this possibility (browser only) should be much more clearly presented (front and center) to potential adopters. To me at least, it's a game changer!
Thank you for your work and time!

PS: I'm not speaking about running it on the client computer (electron as an example) but on the user's BROWSER!

**Comment by planger** [comment]
Hi,

unfortunately, we don't have dedicated documentation yet in GLSP. We hope to be able to add more documentation / examples on that topic after the 2.0 release in August.
What we have so far is that we've restructured and separated the entry points in the Typescript server and the VS Code integration:
- https://github.com/eclipse-glsp/glsp-server-node/tree/main/packages/server/src
- https://github.com/eclipse-glsp/glsp-vscode-integration/tree/master/packages/vscode-integration/src

Both have a `node`, `browser` and `common` folder. For a browser-only deployment the entry point is in `browser` and for a nodejs-based deployment, the entry point is in `node`. The code that is shared is in the `common` folder.

As an adopter you can define via the import which entry point you want to use:
- from `@eclipse-glsp/server` => default, node
- from `@eclipse-glsp/server/browser` => browser entry point
- from `@eclipse-glsp/server/node` => node entry point

We don't have a standalone browser-only example yet. But we do have a VS Code web extension example:
https://github.com/eclipse-glsp/glsp-vscode-integration/tree/master/example/workflow/web-extension

I hope this helps a bit as a starting point. Please let us know if you have specific questions later on.
